src/main/scala/org/clulab/wm/eidos/EidosSystem.scala

```diff
@@ -28,140 +24,153 @@ class EidosSystem(val components: EidosComponents) {
   def annotate(text: String): Document = {
-    val tokenized = components.proc.mkDocument(text, keepText = true) // Formerly keepText, must now be true
-    val annotated = annotateDoc(tokenized)
+    val tokenizedDoc = components.proc.mkDocument(text, keepText = true) // Formerly keepText, must now be true
+    val annotatedDoc = annotateDoc(tokenizedDoc)
 
-    annotated
+    annotatedDoc
   }
 
   // ---------------------------------------------------------------------------------------------
   //                                 Extraction Methods
   // ---------------------------------------------------------------------------------------------
 
-  def extractFrom(doc: Document): Vector[Mention] = {
+  def extractMentionsFrom(doc: Document): Seq[Mention] = {
 
-    def extractEventsFrom(doc: Document, state: State): Vector[Mention] = {
-      val extractedEvents = components.engine.extractFrom(doc, state).toVector
-      val mostCompleteEvents = components.actions.keepMostCompleteEvents(extractedEvents, State(extractedEvents)).toVector
+    def extractMentions(state: State): Seq[Mention] = {
+      val extractedEvents = components.engine.extractFrom(doc, state)
+      val mostCompleteEvents = components.actions.keepMostCompleteEvents(extractedEvents, State(extractedEvents))
 
       mostCompleteEvents
     }
 
     require(doc.text.isDefined)
-    // Prepare the initial state -- if you are using the entity finder then it contains the found entities,
-    // else it is empty
+    // Prepare the initial state.  If you are using the entity finder, then it contains the found entities;
+    // otherwise, it is empty.
     val initialState = components.entityFinders.foldLeft(new State()) { (state, entityFinder) =>
       val mentions = entityFinder.find(doc, state)
+
       state.updated(mentions)
     }
 
-    // Run the main extraction engine, pre-populated with the initial state
-    val events = extractEventsFrom(doc, initialState).distinct
+    // Run the main extraction engine, pre-populated with the initial state.
+    val events = extractMentions(initialState).distinct
     // Note -- in main pipeline we filter to only CAG relevant after this method.  Since the filtering happens at the
-    // next stage, currently all mentions make it to the webapp, even ones that we filter out for the CAG exports
+    // next stage, currently all mentions make it to the webapp, even ones that we filter out for the CAG exports.
     //val cagRelevant = keepCAGRelevant(events)
 
     events
   }
 
-  def postProcess(annotatedDocument: AnnotatedDocument): AnnotatedDocument = {
-    val lastAnnotatedDocument = postProcessors.foldLeft(annotatedDocument) { (nextAnnotatedDocument, postProcessor) =>
-      postProcessor.process(nextAnnotatedDocument)
+  def preProcess(preProcessors: Seq[PreProcessing], odinMentions: Seq[Mention]): Seq[Mention] = {
+    val lastOdinMentions = preProcessors.foldLeft(odinMentions) { (prevOdinMentions, preProcessor) =>
```

Comment:
ok, I think this is clearly elegant, but any chance you can add a comment for the people on the team (with a range of scala experience) that have a hard time reading this syntax (including me!)
If I understand correctly, here we're taking the odinMentions and applying a each of the given preProcessing steps to each mention -- sequentially?
Perhaps we can include some examples in the comments too -- of what "preprocessing" on an odin mention might could look like?

Reply:
Good point and a better name would help.  How about a MentionFilter for Pre and AnnotatedDocumentFilter for Post?  No.  Something with Stage?  The previous method, as far as I could tell, was just a series of prevMentions -> filter -> currMentions -> filter -> nextMentions filter -> ...

Reply:
Maybe it could be something like odinMentionPostProcessors and eidosMentionPostProcessors?
That's a mouthful, but really they're both post-processing more than filtering, but some are done before converting to an eidosMention and some post -- but we can also keep the name as is, but just add a comment to this effect?
It really is much much cleaner!

Reply:
thanks for doing all this!